test_gb_index_select.py

```diff
@@ -0,0 +1,192 @@
+import itertools
+import os
+import time
+from enum import Enum
+
+import dgl
+import dgl.graphbolt as gb
+import torch
+from dgl.data import AsNodePredDataset
+from dgl.dataloading import DataLoader, NeighborSampler
+from ogb.nodeproppred import DglNodePropPredDataset
+
+
+def get_ogbn_graph_test_indices(name):
+    dataset = AsNodePredDataset(DglNodePropPredDataset(name))
+    sampler = NeighborSampler(
+        [10, 10, 10],  # fanout for [layer-0, layer-1, layer-2]
+    )
+    train_idx = dataset.train_idx
+    g = dataset[0]
+    train_dataloader = DataLoader(
+        g,
+        train_idx,
+        sampler,
+        batch_size=1024,
+        shuffle=True,
+        drop_last=False,
+        num_workers=0,
+    )
+    ret_indices = []
+    for it, (input_nodes, output_nodes, blocks) in enumerate(train_dataloader):
+        ret_indices.append(input_nodes)
+    feature = torch.rand(g.number_of_nodes(), g.ndata["feat"].shape[1]).float()
+    return feature, ret_indices
+
+
+class Device(Enum):
+    GPU = 0
+    Pinned = 1
+
+
+def gen_random_testcase(n_rows, feat_size, num_indices):
+    feature = torch.rand(n_rows, feat_size).float()
+    indices = []
+    for i in range(10):
+        indices.append(torch.randint(0, n_rows, (num_indices,)))
+    return feature, indices
+
+
+def test_index_select_throughput(
+    feature, indices, feature_device, indices_device
+):
+
+    feature = (
+        feature.cuda() if feature_device == Device.GPU else feature.pin_memory()
+    )
+    indices = [
+        index.cuda() if indices_device == Device.GPU else index.pin_memory()
+        for index in indices
+    ]
+
+    # Warm up
+    for _ in range(3):
+        for index in indices:
+            torch.ops.graphbolt.index_select(feature, index)
+    torch.cuda.synchronize()
+    n_iter = 5
+    start = time.time()
+    for _ in range(n_iter):
+        for index in indices:
+            torch.ops.graphbolt.index_select(feature, index)
+    torch.cuda.synchronize()
```

Comment:
The use of cuda events here would give more reliable results, especially for smaller test cases. But for larger ones, it should be good enough.